fix(backend): make uniqueness constraint query use indexes

fatih-acar opened the pull request:
Use more granular subqueries depending on the parameters so that the planner can find the best index.
Otherwise the planner might use the attribute name index instead of the attribute value index when the latter is available.

fatih-acar committed:
fix(backend): make uniqueness constraint query use indexes

Use more granular subqueries depending on the parameters so that the
planner can find the best index.
Otherwise the planner might use the attribute name index instead of the
attribute value index when the latter is available.

Signed-off-by: Fatih Acar <fatih@opsmill.com>

fatih-acar commented:
Please do not merge yet, I need to run benchmarks with the feature flag disabled to confirm there's no regression.

ajtmccarty reviewed (approved):
this looks right to me

fatih-acar committed:
fix(backend): add extra where clause in uniqueness constraint query

This should prevent potential performance regression when the value
index is not present (feature flag disabled).

Signed-off-by: Fatih Acar <fatih@opsmill.com>
